src/Operations.jl

```diff
@@ -1546,7 +1574,9 @@ function pin(ctx::Context, pkgs::Vector{PackageSpec})
     foreach(pkg -> update_package_pin!(ctx.registries, pkg, manifest_info(ctx.env.manifest, pkg.uuid)), pkgs)
     pkgs = load_direct_deps(ctx.env, pkgs)
 
-    pkgs, deps_map = _resolve(ctx.io, ctx.env, ctx.registries, pkgs, PRESERVE_TIERED, ctx.julia_version)
+    # TODO: maybe this needs to allow PRESERVE_ALL_INSTALLED (disabled with false)
+    pkgs, deps_map = _resolve(ctx.io, ctx.env, ctx.registries, pkgs, PRESERVE_TIERED, ctx.julia_version, false)
```

Comment:
Imo we should change `pin` to not take a version and just have it pin on the current version. Then there is no need to resolve after a `pin`.

Reply:
Sounds good. Another PR?